### PR TITLE
chore: Refactored test to React

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR introduces the client test

Blocked by #26 #25 #24 #23

### CODE REVIEW

File: __react/template/src/tests/$$channel$$.spec.ts.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+export default function channelRender({ channelName, channel, params }) {
+  return <File name={`${pascalCase(channelName)}.spec.ts`}>
+    {`
+import {describe, it, before} from 'mocha';
+import {expect} from 'chai';
+import * as Client from '../index'
+import * as TestClient from './testclient/index'
+import { NatsTypescriptTemplateError } from '../NatsTypescriptTemplateError';
+
+describe('${channelName} can talk to it self', () => {
```
Comment (fmvilas) [inline __react/template/src/tests/$$channel$$.spec.ts.js:63]:
```suggestion
describe('${channelName} can talk to itself', () => {
```

**Review by fmvilas — DISMISSED** [review]
Great stuff! 👏

**Review by magicmatatjahu — APPROVED** [review]

**Comment by jonaslagoni** [comment]
Forgot to add the test components for request/reply and publish/subscribe, also have removed any duplicate code 😄

**Review by magicmatatjahu — APPROVED** [review]
